### PR TITLE
Remove styled-components (closes #56)

### LINKED ISSUE #56: [feat] Remove styled-components

### PR DESCRIPTION
**Pull request by Gustl22**
### Fix issues

- [x] #56

### Description

- Remove styled-components
   - Also adapted docs accordingly
   - Additionally kept classnames, just in case someone uses these in an extra stylesheet.
- Update storybook
    - new addon for themes needed
    - Some rewrites due to: https://github.com/react-theming/storybook-addon-material-ui/issues/126#issuecomment-790733030
- Fix and document tests
    - Some tests for `ColorButton` cannot performed completely, most likely due to: https://github.com/jsdom/jsdom/issues/1696 and https://github.com/facebook/jest/issues/8464
    - See also explanation in code, I really spent days to find a solution, also tried Enzyme, like described in [MUI Testing](https://material-ui.com/guides/testing/#createmount-options-mount). But this only refers to the properties react components themselves, and don't consider the styles of classes in the rendered result I handed over one step before. And I think it doesn't make any sense to test properties of a component, although the actual css style, which is displayed is the critical part to check. This may comes down to visual testing (?), if it's worth the effort.
    - I also went through the "solution" / code of `styled-components` and they actually save their style-rules for each class and look them up if needed, e.g. when testing. As far as I understood they actually not loading the style rules from the DOM.

I actually only went through the storybook and checked if anything jumped into my eye. Not sure if I covered all cases a 100%.

### Check List

- [x] respect code conventions and usages
- [ ] tests and 100% coverage, (yes, but depends, more a wontfix issue)
- [x] well documented
- [x] self review

### Review targets:
- tests / storybook / styles

### CODE REVIEW

**Comment by mikbry** [comment]
Thanks @Gustl22 for your hard work! I review your PR ASAP

**Review by mikbry — CHANGES_REQUESTED** [review]
From my tests, I see one change: button stay focused and highlighted which was not the case previously. See the blue highlight
Screenshot 2021-03-09 at 13 37 16
Instead of
Screenshot 2021-03-09 at 13 39 34
Compare using:
https://mikbry.github.io/material-ui-color/?path=/story/components-colorpalette--basic

File: src/components/ColorPicker.jsx [hunk 1]
```diff
@@ -19,15 +18,18 @@ import * as ColorTool from '../helpers/colorTool';
 import uncontrolled from '../helpers/uncontrolled';
 import * as CommonTypes from '../helpers/commonTypes';
 import useTranslate from '../helpers/useTranslate';
+import { makeStyles } from '@material-ui/core/styles';
```
Comment (mikbry) [inline src/components/ColorPicker.jsx:21]: `@material-ui/core/styles` import should occur before import of `./ColorButton`

File: src/components/ColorBox/index.jsx [hunk 2]
```diff
@@ -18,77 +17,71 @@ import ColorInput from '../ColorInput';
...
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+    padding: '8px 4px 8px 8px',
+    justifyContent: 'space-between',
+  },
+  colorboxInput: {},
+  colorboxColorBg: {
+    width: 48,
+    height: 48,
+    background: 'linear-gradient(45deg, #ccc 25%, transparent 25%), linear-gradient(135deg, #ccc 25%, transparent 25%), linear-gradient(45deg, transparent 75%, #ccc 75%), linear-gradient(135deg, transparent 75%, #ccc 75%)',
```
Comment (mikbry) [inline src/components/ColorBox/index.jsx:59]: Insert `⏎·····`eslintprettier/prettier

**Review by mikbry — CHANGES_REQUESTED** [review]
Some linting/prettier errors i got using VSCode

**Comment by Gustl22** [comment]
Sorry I'll make the changes the upcoming week. Thx for reviewing :)

File: src/components/ColorPicker.jsx (hunk 1 above)
Comment (Gustl22) [reply src/components/ColorPicker.jsx]: Fixed

File: src/components/ColorBox/index.jsx (hunk 2 above)
Comment (Gustl22) [reply src/components/ColorBox/index.jsx]: Fixed

**Comment by Gustl22** [comment]
> From my tests, I see one change: button stay focused and highlighted which was not the case previously. See the blue highlight
> https://mikbry.github.io/material-ui-color/?path=/story/components-colorpalette--basic

Sorry can you explain more detailed? On your mentioned deployed storybook colorpalette, I also get a highlight after clicking on it.
Feel free to alter my code, if don't want to wait that long :)

**Review by mikbry — APPROVED** [review]

**Comment by mikbry** [comment]
I just tested again and all is fine ;-)

**Comment by mikbry** [comment]
There is one conflict in yarn.lock, do you fix it or i do ? after that i will merge

**Comment by Gustl22** [comment]
Sure, I can do it quickly :)

**Comment by mikbry** [comment]
Ok, do it ;-)

**Comment by Gustl22** [comment]
Sorry, I found one more issue with the transparent palette. I'll let you know, when ready :)

**Comment by Gustl22** [comment]
Apparently styled-components behave differently from makestyles, when using undefined value (remove attribute vs. leave old value), so I set a default value for background-color and background-image.

I also completely refreshed the yarn.lock, if thats ok, in order to not mess up the previous conflicts.

**Review by mikbry — APPROVED** [review]

**Comment by mikbry** [comment]
Well done, thanks @Gustl22 !

**Comment by mikbry** [comment]
Ouch I found a bug in this PR, displaying erroneous color:
in ColorPicker 
- change hex textfield to 'xxx'
- the red error is mispositionned
Screenshot 2021-03-16 at 16 46 57

**Pull request merged**